File: github_jira_tools/github_tags.py
import click
from github import Github

@click.command()
@click.option('--token', '-t', prompt='Your access token', help='The token used to authenticate the user.')
@click.option('--repo', '-r', prompt='Repository', help='The repository which contains the issues. E.g. mattermost/mattermost-server')
@click.option('--labels', '-l', prompt='Labels', help='The labels to set to the issues', multiple=True)
@click.argument('issue-numbers', nargs=-1)
def cli(token, repo, labels, issue_numbers):
    if len(issue_numbers) < 1:
        print("You need to pass at least one issue number")
        return

    g = Github(token)
    r = g.get_repo(repo)
    final_labels = []
    for label in r.get_labels():
        if label.name in labels:
            final_labels.append(label)

    for issue_number in issue_numbers:
        try:
            issue = r.get_issue(int(issue_number))
            issue.set_labels(*final_labels)
        except Exception as e:
            print("Unable to update issue {}. error: {}".format(issue_number, e))

if __name__ == "__main__":
    cli()

File: github_jira/cmd/githubtags.go
package cmd

import (
	"errors"
	"fmt"
	"strconv"

	ghu "github.com/mattermost/mattermost-utilities/github_jira/github_utils"
	"github.com/spf13/cobra"
)

var GithubLabelsCmd = &cobra.Command{
	Use:     "labelgithub",
	Short:   "Add labels to github issues provided in list of issue-numbers",
	Example: `  labelgithub -t <github token> -r mattermost/mattermost-server -l 'Tech/Go,Help Wanted,Up For Grabs' 19977 12345`,
	Args:    cobra.MinimumNArgs(1),
	RunE:    createGithubLabels,
}

func init() {
	GithubLabelsCmd.Flags().StringP("token", "t", "", "The token used to authenticate the user against github.")
	GithubLabelsCmd.MarkFlagRequired("token")
	GithubLabelsCmd.Flags().StringP("repo", "r", "mattermost/mattermost-server", "The repository which contains the issues. E.g. mattermost/mattermost-server")
	GithubLabelsCmd.MarkFlagRequired("repo")
	GithubLabelsCmd.Flags().StringSliceP("labels", "l", []string{}, "The labels to set to the issues")
	GithubLabelsCmd.MarkFlagRequired("labels")
	RootCmd.AddCommand(GithubLabelsCmd)
}

func createGithubLabels(command *cobra.Command, args []string) error {

	ghToken, err := getNonEmptyString(command, "token")
	if err != nil {
		return err
	}
	repo, err := getNonEmptyString(command, "repo")
	if err != nil {
		return err
	}
	ghRepo, err := ghu.ParseRepo(repo)
	if err != nil {
		return err
	}

	labels, err := getNonEmptyStringSlice(command, "labels")
	if err != nil {
		return err
	}

	var intArgs []int

	for _, arg := range args {
		i, err := strconv.Atoi(arg)
		if err != nil {
			fmt.Println(err)
		}
		intArgs = append(intArgs, i)
	}

	client := ghu.GetClient(ghToken)

	// Get valid labels
	validLabels, errLabels := ghu.GetLabelsList(client, ghRepo, labels)
	if errLabels != nil {
		return errLabels
	}

	if len(validLabels) == 0 {
		return errors.New("no matching labels were found")
	}
	fmt.Println(intArgs)
	validIssues, errIssues := ghu.GetIssuesList(client, ghRepo, intArgs)
	if errLabels != nil {
		return errIssues
	}

	if len(validIssues) == 0 {
		return errors.New("no matching issues were found")
	}

	multiError := ghu.SetLabels(client, ghRepo, validLabels, validIssues)
	var newErr error
	if len(multiError) > 0 {
		newErr = errors.New("multiple errors found")
	}

	for mErr := range multiError {
		fmt.Println(multiError[mErr])
	}

	return newErr
}
